examples/advanced/tidb-cluster.yaml

```diff
@@ -1,295 +1,872 @@
+    #       - "preStop"
+
+    ## Affinity for pod scheduling
     ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
     # affinity:
     #   podAntiAffinity:
     #     preferredDuringSchedulingIgnoredDuringExecution:
-    #       - podAffinityTerm:
-    #           labelSelector:
-    #             matchExpressions:
-    #               - key: app.kubernetes.io/component
-    #                 operator: In
-    #                 values:
-    #                   - pd
-    #                   - tikv
-    #           topologyKey: ""
-    #         weight: 100
-    ## If you use require affinity, you must ensure that at least 3 nodes are available in the cluster
-    #     requiredDuringSchedulingIgnoredDuringExecution:
-    #       - labelSelector:
+    #     - podAffinityTerm:
+    #         labelSelector:
     #           matchExpressions:
-    #             - key: app.kubernetes.io/component
-    #               operator: In
-    #               values:
-    #                 - tidb
-    #         topologyKey: ""
-
-    ## Specify node labels for pods assignment
-    ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
-    # nodeSelector:
-    #   app.kubernetes.io/component: tidb
+    #           - key: app.kubernetes.io/component
+    #             operator: In
+    #             values:
+    #             - pd
+    #             - tikv
+    #         topologyKey: kubernetes.io/hostname
+    #       weight: 100
+    #     # require not to run TiDB pods on nodes where there's already a TiDB pod running
+    #     # if setting this, you must ensure that at least `replicas` nodes are available in the cluster
+    #     requiredDuringSchedulingIgnoredDuringExecution:
+    #     - labelSelector:
+    #         matchExpressions:
+    #         - key: app.kubernetes.io/component
+    #           operator: In
+    #           values:
+    #           - tidb
+    #       topologyKey: kubernetes.io/hostname
 
   tikv:
-    # Base image of the component
-    baseImage: pingcap/tikv
+    ############################
+    # Basic TiKV Configuration #
+    ############################
 
-    # TiKV Configuration
-    # https://pingcap.com/docs/tidb-in-kubernetes/stable/configure-cluster-using-tidbcluster/#configure-tikv-parameters
-    config: {}
+    ## Base image of the component
+    baseImage: pingcap/tikv
 
-    # The desired ready replicas
+    ## tikv-server configuration
+    ## Ref: https://docs.pingcap.com/tidb/stable/tikv-configuration-file
+    config:
+      storage:
+        # In this examples, we set this to avoid using too much storage.
+        reserve-space: "0MB"
+      rocksdb:
+        # In this examples, we set this to avoid the following error in some Kubernetes clusters:
+        # "the maximum number of open file descriptors is too small, got 1024, expect greater or equal to 82920"
+        max-open-files: 256
+      raftdb:
+        max-open-files: 256
```

Comment:
We should not set this in production env, we'd not better not put the configs may affect the performance or running behavior of the component here, just put some  common configs, like log level to give an example to the users, for the others, leave to the users.